03_analysis/probe_parameter.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python
+
+"""
+probe_parameter.py
+
+For specific force field parameters identified from the analysis of
+tailed_parameters.py, (1) find all molecules that use this parameter,
+and (2) save them to a single file labeled with the parameter ID.
+Then (3) generate a PDF report of all molecules together, color-coded
+by parameter ID and labeled with parameter ID and SMILES tag.
+
+By:      Victoria T. Lim
+Version: Feb 6 2020
+
+Examples:
+$ python probe_parameter.py -f openff_unconstrained-1.0.0-RC2.offxml -k tailed_rmsd.pickle -s probe_params_rmsd -p a20 b9 n3 t96
+
+"""
+
+import os
+import pickle
+
+import openeye.oechem as oechem
+import openeye.oedepict as oedepict
+
+from openforcefield.topology import Molecule, Topology
+from openforcefield.typing.engines.smirnoff import ForceField
+
+
+### ------------------- Functions -------------------
+
+
+def probe_by_parameter(probe_param, ffxml, subdir, all_probe_mols, inpickle):
+    """
+    For a single force field parameter, (1) find all molecules that use this
+    parameter, and (2) save them to a mol2 file labeled with the parameter ID.
+
+    Parameters
+    ----------
+    probe_param : string
+        Name of the parameter to investigate
+    ffxml : string
+        Name of the FFXML force field file
+    subdir : string
+        Name of subdirectory in which to save mol2 files for each parameter
+    all_probe_mols : dict
+        key is string of a parameter id to be probed;
+        value is an empty list
+    inpickle : string
+        Name of the pickle file from output of tailed_parameters.py
+
+    Returns
+    -------
+    all_probe_mols:
+        key is string of a parameter id to be probed;
+        value is a list of oegraphmols with this parameter id
+    """
+    prefix_dict = {'a':'Angles', 'b':'Bonds', 'i':'ImproperTorsions', 'n':'vdW', 't':'ProperTorsions'}
+
+    # load parameter dictionaries from pickle
+    with open(inpickle, 'rb') as f:
```

Comment:
Pickle isn't very stable with respect to Python versions. Consider looking into automatically serializable representations such as Pydantic.